### ISSUE TITLE
.DS_Store open by default?

### ISSUE THREAD

**Issue by dragonxlwang** [Osmose/advanced-open-file#96]
Recently after upgrading to 0.14.3 I found that advanced-open-file will always open ".DS_Store" file of the directory by default (I am using a Mac). How can I disable it? This won't happen when using default open file in atom though. Thanks,

![image](https://cloud.githubusercontent.com/assets/15812510/13613598/70f0627e-e532-11e5-8c8f-1e369636205a.png)

**Comment by Osmose** [comment]
Thanks for the report!

When is it opening the file? While you're browsing, it just opens automatically? Or is it when you try to open another file?

Basically, what are your steps to reproduce?

**Comment by dragonxlwang** [comment]
yes. For example:
1. open adv-open-file by alt+cmd+o
2. type "~/workspace/foo.bar"
3. it will open both ~/workspace/foo.bar and ~/workspace/.DS_Store. The later one opened automatically. 

My current temp fix is to delete all .DS_Store in workspace recursively...

**Comment by Osmose** [comment]
That's really weird; I'm unable to replicate this. I'm also on OSX.

It might be a weird interaction with another package. Does this problem persist if you disable all your other installed packages temporarily?

**Comment by dragonxlwang** [comment]
yes. I am having this problem with only the advanced-open-file packages… Not sure if this is related to OS X instead…?

**Comment by dragonxlwang** [comment]
I realized maybe this is due to that I set `defaults write com.apple.finder AppleShowAllFiles YES` a while ago. However, not sure if that is indeed the cause.

**Comment by Osmose** [comment]
I'm still unable to replicate this. My system setting for `com.apple.finder AppleShowAllFiles` is `1`, which I believe is equivalent to your value. At least, my Finder shows hidden files fine.

If you create a directory with two files in it, `.DS_Store` and `foo.bar`, and try to open `foo.bar`, does it still auto-open `.DS_Store` as well? If not, what if you delete `.DS_Store`, create an image file within the directory, and then view it in Finder to generate a real `.DS_Store` file? 

Basically I'm try to get an isolated test case that I can play around with replicating. Any help you can give is appreciated! :D

**Comment by ripter** [comment]
I was unable to replicate the issue. I recored my attempt: https://youtu.be/zi8PPT0H5Hc
@dragonxlwang is there something I'm doing differently?

**Comment by pirminis** [comment]
this issue is present for me too.

further are instructions.

requirements:
 - `atom-beta@1.9.0-beta0`
 - `advanced-open-file@0.16.3`
 - example root path for project: `/a/b/c`

Options:
 - [x] Create directories
 - [ ] Create files instantly
 - Default input value – `Project root`
 - [x] Use fuzzy matching for matching filenames
 - [ ] Shortcuts for fast  directory switching

Then:
 - open advanced open file dialog using command pallete (`cmd-shift-p`)
 - IMPORTANT: if root path is `/a/b/c`, and directory `/a/b/c/d/e` exists, then new file should be `/a/b/c/d/e/newfile.here`. Creating new file in root path doesn't reproduce this bug
 - press enter, new file is created. `.DS_Store` file is opened automatically too.

Noted:
 - automatically opened `.DS_Store` file resides in root path, aka `/a/b/c/.DS_Store`

**Comment by Osmose** [comment]
@pirminis I followed your steps to the letter, same atom version, same package version, same settings, etc. and am still unable to replicate the issue. Could you find a directory tree where you consistently experience the issue and then share the filenames in that tree? The out of the `tree` command would be ideal.

I'm wondering if it has something to do with the directory layout that's confusing the fuzzy finding code.
